src/cargo/ops/mod.rs

```diff
@@ -49,6 +49,7 @@ mod cargo_run;
 mod cargo_test;
 mod cargo_uninstall;
 mod common_for_install_and_uninstall;
+pub mod edit;
```

Comment:
> Where should the editing API should live? 

> Justification: precedent has been set by cargo::ops::resolve and others to have utils shared by multiple ops live in cargo::ops. This is also serves to be a rather conservative API change.

I guess this depends on what justifies code being an `op` vs ` util`.

Personally, I see this as low level like the manifest loading and would put it in a `util::toml_mut` to match `util::toml`.

I don't have too strong of an opinion and am willing to wait in case anyone else has any other thoughts.

Reply:
I am concerned that `util::toml_mut` is not distinct or descriptive enough. The other problem would be confusing `util::toml_mut::Manifest` and `util::toml::Manifest`. I could see something similar to `util::toml::parse` and `util::toml::edit` but `util::toml::edit` does parse the file. It also doesn't resolve  `util::toml::edit::Manifest` and `util::toml::parse::Manifest`. I'm not sure if I have a suggestion that resolves everything but I agree that it should probably be moved into `util`.

Reply:
> I am concerned that util::toml_mut is not distinct or descriptive enough. 

`util::toml` isn't either.  I was hoping to change these later but focus on consistency for now.

> It also doesn't resolve util::toml::edit::Manifest and util::toml::parse::Manifest

So far these have been used in distinct enough code paths to not be a problem.  While this does raise visibility of these types, I would personally punt on the naming.

If people want to handle the naming, I would just do `Mut` suffixes on them (`ManifestMut`, `DependencyMut`).

Reply:
I'm happy if a consensus is reached on `util::toml` and `util::toml_mut`, but I think it's awkward to have such similar names given that the two barely overlap in functionality. That's why I tend to prefer "edit" as the distinguishing name, particularly since projects like `toml_edit` have already established that convention.

> I could see something similar to `util::toml::parse` and `util::toml::edit` but `util::toml::edit` does parse the file.

`util::toml::parse` (or perhaps `util::toml::read`) makes sense to me alongside `util::toml::edit`. Although `util::toml::edit` does parse the file in some sense, that's not how it's intended to be used, and this naming convention makes the distinction more clear.

Reply:
I like `util::toml::read` and `util::toml::edit`. It tells the use case and is distinct. When thinking more about having both `Manifest` and `Dependency` be the same thing, `read` and `edit` also solves that. You want an `edit::Manifest` when editing and a `read::Manifest` when reading.

Reply:
I'm not a fan of nesting further under `util::toml`
- Deep nesting in an API can be a pain to deal with
- As was mentioned, I feel the disjoint nature of these two APIs justifies them being in more clearly separate modules.

Reply:
Longer term, I am hoping to do wider crate re-organization.  This is blocked on moving the architectural documentation into the crate so we can know when we break links.

In the short term, let's go with `util::toml_mut` as it has minimal depth, minimal edit to existing logic, and consistency with exiting logic.